### PR TITLE
Yolov7-Changed eps value

### PR DESCRIPTION
**Pull request by AMIYAMAITY**
Hi, 

Last couple of days I was focusing on getting the correct confidence value and the same detection box in comparison to the WongKinYiu/yolov7 training codebase.

I was create a issue last time regarding the same issue link is - https://github.com/wang-xinyu/tensorrtx/issues/1438

This time I have experimented with one minor thing which is EPS value. I changed `1e-5` to `1e-3`. In my case it's successfully working with the default yolov7 model as well as our custom trained model. After this change in tensorrtx giving exact same detection boxes and confidence as default/custom model gives in pytorch codebase. 

Added some images below-

This inference image from pytorch-
image

This inference image from tensorrtx with eps=1e-3 
image

This inference image from tensorrtx with eps=1e-5
image

See the images, eps of `1e-5` is not good in both detection and confidence value. I attached only one image here but I have tried more than `10k` images. False positives were huge with eps of `1e-5` but after I changed to eps `1e-3` now false positives are reduced, now It's closed to as pytorch inference FPs result.

### CODE REVIEW

**Comment by wang-xinyu** [comment]
@AMIYAMAITY Thanks for digging into this issue, can you also attach the link of yolov7 where it's using eps=1e-3?

**Comment by AMIYAMAITY** [comment]
Hi @wang-xinyu , 

I have not seen `1e-3` while calling `BatchNorm2d` [link](https://github.com/WongKinYiu/yolov7/blob/a207844b1ce82d204ab36d87d496728d3d2348e7/models/common.py#L104). It's taking default value, which is `1e-5` [link](https://github.com/pytorch/pytorch/blob/82099ab87bb789670cd95c91fac0681958f3be94/torch/nn/modules/batchnorm.py#L129C1-L130C1).  


But when I compared the function definition with yolov5, I have seen it in [tensorrtx/yolov5](https://github.com/wang-xinyu/tensorrtx/blob/4dae28be209c22dace596ab06895de7914f86d69/yolov5/src/model.cpp#L108) here. Not sure why it had used that time as well. Seems like `1e-3` is more stabilized than `1e-5` in tensorrtx.

**Comment by wang-xinyu** [comment]
Emm, then theoretically, we should use 1e-5. There must be other reason which was causing the diff.  Still, you should try to compare mAP with larger dataset, not just couple of images.

yolov5 itself is using 1e-3, tensorrt implementation should always align with the original network structure.

**Comment by AMIYAMAITY** [comment]
Hi @wang-xinyu ,

Yes technically we should use `1e-5`, since it's there in the pytorch code base of `yolov7`. Actually I have analysed over 10.5k images, and analysis against ground truth. I am getting below precision and recall numbers with `1e-5` eps.
image

But after changing the fps `1e-5` to `1e-3` it got resolved seems like. Below the snapshot of precision and recall metrics used eps= `1e-3`
image

All the above snapshot inference from tensorrtx codebases-

Below snapshot inference from pytorch -
image



Another doubt I have is, I went through the [ultralytics/yolov5](https://github.com/ultralytics/yolov5/tree/master) code base of yolov5. Here I can see [BatchNorm2d](https://github.com/ultralytics/yolov5/blob/574331f984c0aa9c26c4ea78dac90133cfe6b2d0/models/common.py#L81) is calling with default eps value which is `1e-5`. But for [tensorrtx/yolov5](https://github.com/wang-xinyu/tensorrtx/blob/4dae28be209c22dace596ab06895de7914f86d69/yolov5/src/model.cpp#L108) using 1e-3 for eps. Not sure why in both cases like `tensorrtx/yolov5` and `tensorrtx/yolov7` working fine with `1e-3`.

**Comment by wang-xinyu** [comment]
> Another doubt I have is, I went through the [ultralytics/yolov5](https://github.com/ultralytics/yolov5/tree/master) code base of yolov5. Here I can see [BatchNorm2d](https://github.com/ultralytics/yolov5/blob/574331f984c0aa9c26c4ea78dac90133cfe6b2d0/models/common.py#L81) is calling with default eps value which is `1e-5`. But for [tensorrtx/yolov5](https://github.com/wang-xinyu/tensorrtx/blob/4dae28be209c22dace596ab06895de7914f86d69/yolov5/src/model.cpp#L108) using 1e-3 for eps. Not sure why in both cases like `tensorrtx/yolov5` and `tensorrtx/yolov7` working fine with `1e-3`.


For the doubt about yolov5, can you load the pretrained model in pytorch and print the layers? I guess the pretrained model is using 1e-3.

**Comment by WuxinrongY** [comment]
This is yolov7-tiny model info before fused. And '1e-3' is used in BN layer.image

**Comment by wang-xinyu** [comment]
@WuxinrongY What about the eps value of yolov7.pt?

**Comment by wang-xinyu** [comment]
OK, thanks guys, so we should merge this PR.
Thanks @AMIYAMAITY @WuxinrongY
